src/compas/robots/resources/basic.py

```diff
@@ -70,17 +92,54 @@ def can_load_mesh(self, url):
         is_local_file = len(scheme) in (0, 1) or scheme == 'file'
 
         if is_local_file:
-            return True
+            if os.path.isfile(url):
+                return True
 
         # Only OBJ loader supports remote files atm
         is_obj = _get_file_format(url) == 'obj'
         return scheme in ('http', 'https') and is_obj
 
     def load_mesh(self, url):
+        """Loads a mesh from local storage.
+
+        Parameters
+        ----------
+        url : str
+            Mesh location
+
+        Returns
+        -------
+        :class:`Mesh`
+            Instance of a mesh.
+        """
+        # if url.startswith('file:///'):
+        #     url = url[8:]
+        
+        url = self._get_local_path(url)
+        return _mesh_import(url, url)
+
+    def _get_local_path(self, url):
```

Comment:
Please rename to `_get_mesh_url` , because at this point, it's not defined whether the input is a local path or a remote url.